docs/security/faq.md

```diff
@@ -0,0 +1,188 @@
+  (Only ThunderHub is undeterministic here.)
+* The only other known project that uses declarative approach is `nix-bitcoin`.
+
+## Is it good idea to put my lifesavings into this node?
+
+No.
+
+## Why doesn't this project use Docker?
+
+Docker was **not** designed as a security mechanism and shouldn't be used as such.
+There were actually numerous cases of escaping from Docker containers.
+The Docker daemon adds its own attack surface.
+
+That being said, Docker is just a collection of several mechanisms, some of which may be beneficial for security.
+This project already uses some of them and maybe it will use more in the future (needs to be weighted against disadvantages).
+
+## Does this project use virtual machines?
+
+No, while VMs are a great technology when used properly, such as in the case of Qubes OS, they are heavy.
+It could have negative effect of discouraging users from using it and leaving them with much worse options.
+Finding a sweet spot can be more valuable.
+
+This doesn't mean that VMs will **not** be researched in the future as an optional mechanism.
+It's definitely considered a wishlist item.
+
+## So how does this project isolate services?
+
+Using different (system) users.
+One for each service, except where it makes zero sense.
+
+## Aren't users unsecure? What are the limitation of this design?
+
+There doesn't seem to be any significant issue with using users as an isolation mechanism.
+Android uses this same mechanism and there was no (valid) criticism of such approach.
+(There were other issues unrelated to this particular mechanism.)
+
+Where users probably get bad reputation is from poorly configured systems that have ridiculous errors like leaving bad permissions on a file.
+Another good criticism is bulky privileged APIs expoed to unprivileged processes.
+Such is the case of recent `sudo` vulnerability.
+
+This project mitigates these issues by:
+
+* Automating creation of filesystem items and setting proper permissions by default. This removes the possibility of mistakes.
+* Not depending on bulky APIs and not installing them by default.
+* Using "no new privileges" options which would prevent services from exploiting `sudo` vulnerability.
+
+There's one known issue: possible race conditions when TCP servers are restarted.
+This isn't trivial to exploit.
+Remember, it **still** requires discovering a zero day vulnerability in one of the existing packages!
+Fixing this is planned to be done soon and by using Unix sockets.
+
+## How do I know the software I download is not malware?
+
+The only 100% sure way is to review the project and all its dependencies and building on your own.
+
+If you decide to trust me, then you only need to know my public key fingerprint.
+You can obtain it by cross-referencing several sources ([I'm on Keybase](https://keybase.io/kixunil)) or in person in some cases.
+`apt` already performs signature validation on all installed packages so that part is secure.
```

Comment:
what's the trust in the fingerprint verification here?

Reply:
`apt` automatically trusts all keys in its keyring. In this case those mentioned in setup instructions.